File: BackEnd/Node.js/Wasail/models/product.js
'use strict';

module.exports = (sequelize, Datatypes)=>{
    return sequelize.define('product',{
        product_id:{
            type: Datatypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        product_number:{
            type: Datatypes.INTEGER,
            notNull: true,
        },
        product_name:{
            type: Datatypes.STRING,
            isAlpha: true,
            notNull: true,
        },
        image:{
            type: Datatypes.STRING,
        },
    },{
        underscored:true
    })
}

File: FrontEnd/AppDevelopment/GroceryStoreApp/store/lib/Classes/product.dart
class Product {
  final int productId;
  final String productName;
  final String image;
  final int productNumber;

  Product({
    required this.productId,
    required this.productName,
    required this.image,
    required this.productNumber,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productId: json['product_id'],
      productName: json['product_name'],
      image: json['image'],
      productNumber: json['product_number'],
    );
  }
}